### PR TITLE
digits edits

### LINKED ISSUE #18800: Resolve timeouts in test_digits.py
https://travis-ci.org/sympy/sympy/jobs/659764490?utm_medium=notification&utm_source=github_status
It can be because of the added feature in #18659 which involved random tests.

Ping @sr-murthy

### PR DESCRIPTION
**Pull request by smichr**
#### References to other Issues or PRs

fixes #18800

#### Brief description of what is fixed or changed

Random testing is good when a method is complicated and difficult to verify. When a routine is straightforward and easily verified "smart tests are better than random tests" [Zen of SymPy]

#### Other comments

* `digits` was moved to `digits`
* `digits` now accepts Integers
* docstring was updated to reflect that strings were not being handled by `count_digits`; all numbers are handled by digits (since most numerical work is done with Python numerical literals, not strings); this avoids issues with malformed numbers passed as strings

#### Release Notes

* ntheory
    * `count_digits` works only with numbers (not strings)
    * `digits` is now located in `digits` instead of `factor_`
* utilities
    * `iterables` now includes `is_palindromic` to check if a sequence (or slice thereof) is palindromic

### CODE REVIEW

File: sympy/utilities/iterables.py [hunk 1]
```diff
@@ -19,6 +19,40 @@
     multiset_partitions_taocp, list_visitor, MultisetPartitionTraverser)
 
 
+def is_palindromic(s, i=0, j=None):
+    """return True if the sequence is the same from left to right as it
+    is from right to left in the whole sequence (default) or in the
+    Python slice ``s[i: j]``; else False.
+
+    EXAMPLES
+    ========
```
Comment (oscarbenjamin) [inline sympy/utilities/iterables.py]: This should be capitalised but not all-caps.

File: sympy/utilities/iterables.py [hunk 2]
```diff
@@ -19,6 +19,40 @@
...
+    create a slice of the sequence for testing:
+
+    >>> is_palindromic('abcbb', 1, -1)
+    True
+    >>> is_palindromic('abcbb', -4, -1)
+    True
+
+    See Also
+    ========
+    sympy.ntheory.digits.is_palindromic
```
Comment (oscarbenjamin) [inline sympy/utilities/iterables.py]: I would add a blank line above this.

Also is this a valid sphinx reference? This doesn't match the form in the style guide:
https://docs.sympy.org/latest/documentation-style-guide.html#see-also-section

**Comment by oscarbenjamin** [comment]
This looks good but the release note has an XXX

File: sympy/utilities/iterables.py (hunk 2 above)
Comment (smichr) [reply sympy/utilities/iterables.py]: This was formerly the heading that didn't need a blank line but I can put one in. Regarding the form and style suggestions...it doesn't tell what to do to disambiguate which function you are talking about so this is how I imagine the comment-less singleton list would look like. Am open to other suggestions, however. (For anyone thinking that it should be named something else I quote "this" which says "namespaces are one honking great idea."  The path provides the disambiguation.)

**Comment by smichr** [comment]
> This looks good but the release note has an XXX

I left that there so I would be forced to go back and make the two edits when this is committed. I have to make sure the old note regarding the new `count_digits` doesn't indicate that strings were possible as input.

**Comment by sylee957** [comment]
I see that there was a inconsistency of keywords (`base` vs `b`), and the one in `digits.py` can be changed without deprecation, so you can align that too.

File: sympy/utilities/iterables.py [hunk 3]
```diff
@@ -19,6 +19,40 @@
     multiset_partitions_taocp, list_visitor, MultisetPartitionTraverser)
 
 
+def is_palindromic(s, i=0, j=None):
```
Comment (sylee957) [inline sympy/utilities/iterables.py]: I don't think that there is a usage of `i`, `j` keywords. Slicing can always be done priori, so I don't think that it hurts any generalization without these keywords.

Comment (smichr) [reply sympy/utilities/iterables.py]: I am avoiding having to create a new sequence by allowing the test to be done in place. The call to this routine from `is_palindromic` in `digits` uses the `digits` result which needs to be tested from position 1 through the end.
```python
>>> timeit('[1,2,3,4,5][1:3][0]')
0.5159061519952957
>>> timeit('[1,2,3,4,5][0]')
0.23576541400689166
```

**Comment by smichr** [comment]
I think this is ready once tests pass.

File: sympy/utilities/iterables.py [hunk 4]
```diff
@@ -19,6 +19,42 @@
...
+    """return True if the sequence is the same from left to right as it
+    is from right to left in the whole sequence (default) or in the
+    Python slice ``s[i: j]``; else False.
+
+    EXAMPLES
```
Comment (smichr) [inline sympy/utilities/iterables.py]:
```suggestion
    Examples
```

File: sympy/utilities/iterables.py (hunk 1 above)
Comment (smichr) [reply sympy/utilities/iterables.py]: Ughh...how did this creep into my editing habits!

**Pull request merged**